operator/controllers/loki/lokistack_controller.go

```diff
@@ -298,3 +303,30 @@ func (r *LokiStackReconciler) enqueueForAlertManagerServices() handler.EventHand
 		return requests
 	})
 }
+
+func (r *LokiStackReconciler) enqueueForStorageSecret() handler.EventHandler {
+	ctx := context.TODO()
+	return handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		lokiStacks := &lokiv1.LokiStackList{}
+		if err := r.Client.List(ctx, lokiStacks); err != nil {
+			r.Log.Error(err, "Error getting LokiStack resources in event handler")
+			return nil
+		}
+
+		var requests []reconcile.Request
+		for _, stack := range lokiStacks.Items {
+			if obj.GetName() == stack.Spec.Storage.Secret.Name && obj.GetNamespace() == stack.Namespace {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: stack.Namespace,
+						Name:      stack.Name,
+					},
+				})
+			}
+
+			r.Log.Info("Enqueued requests for all LokiStacks because of Storage Secret resource change", "count", len(requests), "kind", obj.GetObjectKind())
+		}
```

Comment:
A object storage secret belongs to a single `LokiStack` resource as per no sharing of the same bucket across loki instances possible w/o accidentally damaging stored chunk and index files. I suggest to add an early return after appending and address the logger info message accordingly.